[ci] check_link_references failing.

### Describe the problem

It seems that some dependency change in the `check_link_references` process has started breaking with the current setup:


* https://github.com/flutter/website/blob/main/Dockerfile#L63-L73



```
#19 [node 3/3] RUN npm install -g npm firebase-tools@11.0.1
#19 sha256:cacf7c736fcaacba1c7a19ebf56b89259d5cd2e950d40228b4bc6367259dfcb0
#19 0.989 npm ERR! code EBADENGINE
#19 0.991 npm ERR! engine Unsupported engine
#19 0.991 npm ERR! engine Not compatible with your version of node/npm: npm@9.1.1
#19 0.991 npm ERR! notsup Not compatible with your version of node/npm: npm@9.1.1
make[1]: Leaving directory '/home/runner/work/website/website'
#19 0.991 npm ERR! notsup Required: {"node":"^14.17.0 || ^16.13.0 || >=18.0.0"}
#19 0.991 npm ERR! notsup Actual:   {"npm":"8.5.5","node":"v17.9.0"}
#19 0.992 
#19 0.993 npm ERR! A complete log of this run can be found in:
#19 0.993 npm ERR!     /root/.npm/_logs/2022-11-09T22_32_01_529Z-debug-0.log
#19 ERROR: executor failed running [/bin/sh -c npm install -g npm firebase-tools@11.0.1]: exit code: 1
------
 > [node 3/3] RUN npm install -g npm firebase-tools@11.0.1:
------
executor failed running [/bin/sh -c npm install -g npm firebase-tools@11.0.1]: exit code: 1
```

At least 2 PRs are affected similarly:

* https://github.com/flutter/website/pull/7661
* https://github.com/flutter/website/pull/7740


### Expected fix

It _seems_ that the node/npm version that is downloaded by the `Dockerfile` needs to be updated to v18+?

### Additional context

_No response_

--- a/Dockerfile
+++ b/Dockerfile
@@ -64,13 +64,13 @@ RUN dart pub get
 # TODO recommended to move to a SHA based install for security reasons
 FROM flutter AS node
 
-RUN curl -sL https://deb.nodesource.com/setup_17.x -o node_setup.sh && \
+RUN curl -sL https://deb.nodesource.com/setup_18.x -o node_setup.sh && \
       bash node_setup.sh 1> /dev/null
 RUN apt-get update -q && apt-get install -yq --no-install-recommends \
       nodejs \
     && rm -rf /var/lib/apt/lists/*
 # Ensure latest NPM, install global Firebase CLI
-RUN npm install -g npm firebase-tools@11.0.1
+RUN npm install -g npm firebase-tools@11.16.0
 
 
 

